code/game/machinery/doors/door.dm

```diff
@@ -157,6 +157,9 @@ var/list/all_doors = list()
 	if (istype(I, /obj/item/device/detective_scanner))
 		return
 
+	if(isobserver(user) && !isAdminGhost(user))
```

Comment:
I can't attach the picture because GitHub is fucking broken but I talked to Sood in Discord about the logic of this line and he now agrees. tl;dr if I just did `if(!isAdminGhost(user))` it would block everyone EXCEPT admin ghosts.